src/stdatamodels/jwst/datamodels/nirspec_flat.py

```diff
@@ -6,6 +10,27 @@
 __all__ = ['NirspecFlatModel', 'NirspecQuadFlatModel']
 
 
+def _migrate_fast_variation_table(hdulist):
+    # Files produced with NirspecFlatModel and NirspecQuadFlatModel
+    # prior to https://github.com/spacetelescope/stdatamodels/pull/183
+    # have flat_table tables (stored in the FAST_VARIATION extension)
+    # that do not have an "error" column
+    # To allow these older files to load, this function
+    # fills in the missing error column.
+    # We have to iterate over the extensions as the tabl
+    for ext in hdulist:
+        if ext.name != 'FAST_VARIATION':
+            continue
+        # check that table has the required columns
+        # for older files they might be missing an 'err' column
+        table_data = ext.data
+        if ('error', '>f4') not in table_data.dtype.descr:
+            err_column = np.full(table_data.wavelength.shape, np.nan, dtype=[('error', '>f4')])
+            table_data = merge_arrays((table_data, err_column))
```

Comment:
```suggestion
        if 'error' not in table_data.dtype.fields:
            if table_data.dtype['wavelength'].shape:
                dtype = [('error', '>f4', table_data.dtype['wavelength'].shape)]
            else:
                dtype = [('error', '>f4')]
            err_column = np.full(table_data.shape[0], np.nan, dtype=dtype)
            table_data = merge_arrays((table_data, err_column), flatten=True)
```

Reply:
I opened another PR against your branch updating the test based on my comment above: https://github.com/tapastro/stdatamodels/pull/3